### PR TITLE
fix(desktop): 将任务简介中的 Markdown 转为可读纯文本

### PR DESCRIPTION
**Pull request by dashhuang**
### 摘要

任务列表直接显示 Markdown 原文时，长链接会挤掉后面的结论，单行简介可能只剩 `[PR #3943]…`。现在先提取可读正文，再应用 140 字符上限：链接保留标题，去掉格式标记，显示为「PR #3943 已合并，CI 无失败、review 已通过。」

### 变更类型

- [x] `fix` 缺陷修复

### 范围

- 关联 Issue / 需求：用户反馈任务列表简介被 Markdown 链接挤占。
- 本 PR 包含：复用已有 marked 依赖提取纯文本；统一处理历史缓存读取和新消息预览；补充回归用例。
- 明确不包含：开发版区域选择、数据库迁移、消息正文渲染和远程查看端的兼容兜底。
- 用户可见变化：简介保留链接标题、标题/列表正文、代码内容和图片说明，清理格式及地址。
- 是否存在 breaking change：无。

### UI 变化

- 引用的设计规范：`docs/design-rules/DESIGN.md` §§1、3、10、14.1。仅改变预览文字，沿用既有字体、行高、截断及语义颜色，Light/Dark 样式均保留。
- 示例：`[PR #3943](https://github.com/makecindy/cindy/pull/3943) 已合并` → `PR #3943 已合并`。
- 平台：macOS Desktop；用户在共享开发版确认效果。未上传含私人任务内容的截图。

### 自动验证

```text
corepack pnpm test:unit:related
结果：通过。

corepack pnpm --filter desktop run --if-present typecheck
结果：通过。

corepack pnpm --filter desktop exec vitest run src/main/localDb/__tests__/mapperMarkdownPreview.test.ts src/main/localDb/__tests__/mapperCitation.test.ts --reporter=dot
结果：2 个文件、17 个用例通过。
```

### 手工验证

macOS，中国大陆版共享开发实例启动就绪，用户确认效果可提交。

### 未执行的验证

- 未分别记录 Light/Dark 目检证据；本次没有修改主题或布局。
- 未在远程被控端安装验证；远程简介由被控端生成，需要被控端更新后生效。
- 未执行 Windows 实机验证和本地全量单测，完整单测由 CI 执行。

### 风险分类

- [x] 其他：预览文本转换。

### 影响与回滚

- 影响范围：Desktop 生成的最近消息简介。仅转换派生显示内容，不改写消息正文或数据库结构。旧端已经截断丢失的内容无法在查看端恢复；通用 HTML 实体解码不在此次范围内（HTML 图片属性沿用正文已有解码规则）。
- 已接受的边界：保留列表查询现有 512 字符原文提取上限；极长链接可能在解析前被截断，导致简介仍含部分链接或缺失后续结论。Dash 已确认本次接受这一显示限制，不扩展数据库 worker 或缓存机制。
- 回滚 / 降级方式：回退本 PR，即恢复原来的空白合并和截断逻辑，无需数据迁移。

### 提交前检查

- [x] 已 review 完整 diff
- [x] 每个 commit 都带 DCO 签名（`git commit -s`）
- [x] UI 改动已注明引用的设计规范章节
- [x] 未提交凭证、令牌或授权文件
- [x] 已补充必要说明及回归用例
- [x] 已确认测试结果或说明未执行原因

### CODE REVIEW

File: apps/desktop/src/main/localDb/mapper.ts [hunk 1]
```diff
@@ -140,7 +139,8 @@ export function finalizePlainPreview(
   // 实现细节。返回 null 与"无预览"同渲染语义。
   if (isSyntheticTriggerText(next)) return null;
   if (role === 'assistant') next = stripInternalWebCitations(next);
-  const collapsed = next.replace(/\s+/g, ' ').trim();
+  // Strip formatting before the display limit: URLs must not consume the preview budget.
+  const collapsed = markdownPreviewText(next);
```
[the remark this answers is not in the document]
Comment (dashhuang) [reply apps/desktop/src/main/localDb/mapper.ts:143]: 已在 8235e6a91 修复。核查实际 SQL 后，批量 backfill 忽略 payload.preview、从消息重算原文；真正混入已转换文本的是删除路径。现在删除与新增/更新一致：保留事务失效，只返回即时简介，由 SQL 回填原文。新增真实 SQLite 冷读→回填→重读→失效后回填用例，覆盖 fenced code、分隔线代码和转义星号。33 个定向用例、相关单测及 Desktop typecheck 通过。

File: apps/desktop/src/shared/markdownPreviewText.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+function previewTokens(tokens: Token[]): string {
+  return tokens
+    .map((token): string => {
+      switch (token.type) {
+        case 'space':
+        case 'br':
+        case 'hr':
+        case 'html':
```
[the remark this answers is not in the document]
Comment (dashhuang) [reply apps/desktop/src/shared/markdownPreviewText.ts]: 已在 9808a8021 修复：将正文现有 HTML 图片解析函数原样抽到 shared，简介复用它提取 alt；保留来源与单图片节点限制，不回退长 URL。新增图片、实体、无 alt、不支持来源和 script 用例；简介及正文解析测试共 23 个通过，相关单测门禁及 desktop typecheck 通过。

**Comment by MagicLizi** [comment]
@dashhuang 👋 这个 PR 还有 1 条 review conversation 没 resolve(apps/desktop/src/shared/markdownPreviewText.ts),auto-review 因此暂时跳过、没法继续审查 / 合并。

如果你已经按评论改完或回应了,请到对应 thread 上点 **Resolve conversation**;全部 resolve 后,下一轮 auto-review 会自动重新审查这个 PR。

File: apps/desktop/src/shared/markdownPreviewText.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { Lexer, type Token, type Tokens } from 'marked';
+
+/** Extract display text without generating HTML or counting link destinations. */
+export function markdownPreviewText(markdown: string): string {
+  return previewTokens(Lexer.lex(markdown, { gfm: true }))
```
[the remark this answers is not in the document]
Comment (dashhuang) [reply apps/desktop/src/shared/markdownPreviewText.ts:7]: 已与 Dash 确认范围：本 PR 保留现有 512 字符原文提取上限，仅做预览的 Markdown 转纯文本。超长链接可能在解析前被截断、无法保留后续结论，这是本次接受的显示边界，不影响消息正文或数据完整性。无需为此扩展数据库 worker 或新增缓存兼容机制；本线程按范围确认收口，并非宣称已修复该边界。

**Review by zqchris — APPROVED** [review]

**Review by MagicLizi — APPROVED** [review]
## Code Review: PR #4130 — fix(desktop): 将任务简介中的 Markdown 转为可读纯文本

**Tier**: standard | **Head**: 9808a80 | **Reviewer**: magiclizi (auto)

### Summary

This PR converts Markdown-formatted task previews to readable plain text in the Desktop sidebar, using the existing `marked` dependency's `Lexer` for tokenization instead of HTML generation. Key changes:

1. **New `markdownPreviewText.ts`** — Token-based Markdown-to-plain-text extractor that preserves link labels, heading/list/code/table text, and HTML `` alt text while stripping formatting and URLs.
2. **Refactored `htmlImage.ts`** — Moved HTML `` parsing from renderer-only to `shared/` for reuse by both the renderer plugin and the preview extractor. Byte-identical logic extraction.
3. **Integrated into `mapper.ts`** — `finalizePlainPreview` now calls `markdownPreviewText()` instead of raw whitespace collapse, ensuring URLs don't consume the 140-char preview budget.
4. **Removed double-persist in deletion path** — `commitMessageDeletion` no longer calls `persistSessionListPreview` with display text, avoiding double-parsing on subsequent reads (the SQL backfill repopulates with raw Markdown).
5. **Comprehensive tests** — 17+ test cases covering links, structured Markdown, code, images, HTML entities, unsafe sources, malformed input, cache round-trips, and the backfill contract.

### Review Checklist

- [x] No credentials, tokens, or secrets in diff
- [x] No new dependencies (reuses existing `marked ^16.4.2`)
- [x] `htmlImage.ts` extraction is byte-identical to the original in `remarkHtmlImages.ts`
- [x] `ATTR_RE` global regex properly resets `lastIndex` before each use
- [x] `htmlImgToImageNode` called with correct `{ type: 'html', value: token.text }` mapping
- [x] Non-img HTML tokens (script, div, etc.) safely return empty string via `null?.alt ?? ''`
- [x] `finalizePlainPreview` applied consistently on all read paths: `listPreview`, `latestMessageExtract`, `latestMessageContent`
- [x] SQL backfill stores raw Markdown; display conversion happens at read time — no double-parse
- [x] Deletion path still broadcasts preview for immediate UI refresh, just doesn't persist to DB
- [x] `persistSessionListPreview` still exists and is used by `invalidateSessionListPreview` — only the deletion-path usage was removed
- [x] Indentation/formatting changes in `messages.ts` are mechanical (block indent fix, import line wrapping)
- [x] Tests validate the backfill round-trip: raw Markdown → DB → read → finalize → consistent display
- [x] Accepted boundary (512-char SQL extraction limit) explicitly documented in PR and resolved thread
- [x] CI all green (client-ci, pr-design-basis both passed)

### Findings

**0 P0, 0 P1.**

No blocking issues found. The implementation is well-structured, the refactoring is clean, edge cases are covered by tests, and the accepted limitation is properly documented.

### Bot settle notice

copilot-pull-request-reviewer and chatgpt-codex-connector timed out (waited >187 min). Their opinions may arrive after merge. Codex did complete its review earlier on the same head with no blocking findings.
